### PR TITLE
adding domain to solid mechanics custom boundary integral

### PR DESCRIPTION
**Pull request by kswartz92**
I need to apply a Robin b.c. to a particular boundary attribute so it would be convenient to apply the custom boundary integral to a `serac::Domain`. This PR is very similar to the changes in #1075.

### CODE REVIEW

**Review by jamiebramwell — APPROVED** [review]
Sorry I left this one waiting. Thanks @kswartz92 !

**Comment by jamiebramwell** [comment]
@samuelpmishLLNL @btalamini Any comments on this one?

File: src/serac/physics/tests/solid_robin_condition.cpp [hunk 1]
```diff
@@ -74,13 +74,22 @@ void functional_solid_test_robin_condition()
...
   // clang-format off
+  Domain robinDomain = Domain::ofBoundaryElements(StateManager::mesh(mesh_tag), 
+    [](std::vector<serac::tensor<double, dim>> X, int) {
+      bool inDomain = false;
+      for (const auto& p : X) {
+        if (p[0]<0.01) {
+          inDomain = true;
+        }
+      }
+      return inDomain;
```
Comment (samuelpmishLLNL) [inline src/serac/physics/tests/solid_robin_condition.cpp]:
```suggestion
      for (const auto& point : X) {
        if (point[0]<0.01) { return true; }
      }
      return false;
```

**Review by samuelpmishLLNL — APPROVED** [review]
sorry I forgot about this one, it looks good to me

**Pull request merged**